src/map_io/map_buildingdata_packet.cc

```diff
@@ -1295,7 +1312,13 @@ void MapBuildingdataPacket::write_productionsite(const ProductionSite& productio
 	fw.string(productionsite.statistics_string_on_changed_statistics_);
 	fw.string(productionsite.production_result());
 
-	fw.signed_32(productionsite.main_worker_);
+	if (productionsite.main_worker_ < 0) {
+		fw.unsigned_8(0);
```

Comment:
// The mapping to usigned_8 will map the -1 to 255 so better save as 0

Reply:
This does not save the main_worker_ variable. If main_worker_ is -1, we only write a 0 to indicate that no main_worker_ is set; otherwise we write 1 (indicating that it is set) followed by the main worker's serial (which will be used on loading later to detect this worker's new slot number).

Reply:
So we might actually write boolean true/false as well. OK let have it then